modules/tinymce/src/core/main/ts/textpatterns/keyboard/KeyHandler.ts

```diff
@@ -9,6 +9,11 @@ import { InlinePatternSet, PatternSet } from '../core/PatternTypes';
 import * as Utils from '../utils/Utils';
 
 const handleEnter = (editor: Editor, patternSet: PatternSet): boolean => {
+  // TINY-8779: mceInsertNewLine calls normalize() which alters the structure of the body content,
+  // in which text nodes of a paragraph are merged into one. This makes the range of any matching patterns become invalid
+  // as the text nodes are no longer existed after normalization. So do a normalize() prior finding any matches
+  editor.getBody().normalize();
```

Comment:
This will have a performance impact on the text editor for a large amount of content.

Reply:
Was also told that this solution can't be backported to 5.x as it isn't supported in IE

Reply:
Hmm... it's possible that this isn't working on Safari as it's currently failing with this error 😞 
```
browser.tinymce.core.textpatterns.ReplacementTest / Matches text nodes in a paragraph - Pattern matches the second text node
Expected:
<p><br>no_error</p><p>for sure</p>
Actual:
<p><br>one_error for sure</p><p>&nbsp;</p>
```